vowpalwabbit/gd.cc

```diff
@@ -734,7 +747,7 @@ void save_load_regressor(vw& all, io_buf& model_file, bool read, bool text, T& w
     for (auto it = weights.begin(); it != weights.end(); ++it)
     {
       const auto weight_value = *it;
-      if (*it != 0.f)
+      if (*it != 0.f && weights.is_activated(it.index()))
```

Comment:
add a `//TODO` comment here reminding us to check where else this check is needed